kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiDefaultStream.java

```diff
@@ -0,0 +1,158 @@
+     * @return JSONStreamWriter which would be used to write data to bigquery table
+     */
+    @VisibleForTesting
+    JsonStreamWriter getDefaultStream(String tableName) {
+        return tableToStream.computeIfAbsent(tableName, t -> {
+            boolean tableCreationAttempted = false;
+            int additionalRetriesForTableCreation = 0;
+            Exception mostRecentException;
+            int attempt = 0;
+            do {
+                try {
+                    if (attempt > 0) {
+                        waitRandomTime();
+                    }
+                    return JsonStreamWriter.newBuilder(t, getWriteClient()).build();
+                } catch (Exception e) {
+                    if (BigQueryStorageWriteApiErrorResponses.isTableMissing(e.getMessage())
+                            && !tableCreationAttempted
+                            && getAutoCreateTables()) {
+                        logger.info("Attempting to create table {} ...", tableName);
+                        tableCreationAttempted = true;
+                        // Table takes time to be available for after creation
+                        additionalRetriesForTableCreation = 30;
+                        //TODO: Attempt to create table
+                        logger.info("Table created {} ...", tableName);
+                    } else if (!BigQueryStorageWriteApiErrorResponses.isRetriableError(e.getMessage()) && !tableCreationAttempted) {
+                        logger.error("Failed to create Default stream writer on table {}", tableName);
+                        throw new BigQueryStorageWriteApiConnectException("Failed to create Default stream writer on table " + tableName, e);
+                    }
+                    logger.warn("Failed to create Default stream writer on table {} due to {}. Retry attempt {}...", tableName, e.getMessage(), attempt);
+                    mostRecentException = e;
+                    attempt++;
+                }
+            } while (attempt < (retry + additionalRetriesForTableCreation));
+            throw new BigQueryStorageWriteApiConnectException(
+                    String.format(
+                            "Exceeded %s attempts to create Default stream writer on table %s ",
+                            (retry + additionalRetriesForTableCreation), tableName
+                    ), mostRecentException);
+        });
+    }
+
+    /**
+     * Calls AppendRows and handles exception if the ingestion fails
+     * @param tableName  The table to write data to
+     * @param rows       The records to write
+     * @param streamName The stream to use to write table to table. This will be DEFAULT always.
+     */
+    @Override
+    public void appendRows(TableName tableName, List<Object[]> rows, String streamName) {
+        JSONArray jsonArr = new JSONArray();
+        JsonStreamWriter writer = getDefaultStream(tableName.toString());
+
+        rows.forEach(item -> jsonArr.put(item[1]));
```

Comment:
No, each item is an array of 2 object - first SinkRecord, second JsonObject for that record. We are taking out all JsonObjects are passing in JSOnArray. Both SinkRecord and JSONObject are needed here. SinkRecord would be required for DLQ routing later. 
Sure, the object should definitely exist otherwise it will lead to problems later. I will add check to fail task if the object [1] is not present.

Reply:
Have added a verification step and updated the method description.